src/psyclone/tests/psyir/frontend/fparser2_where_handler_test.py

```diff
@@ -604,19 +525,36 @@ def test_where_mask_containing_sum_with_dim(fortran_reader):
 
 
 @pytest.mark.usefixtures("parser")
-@pytest.mark.parametrize("rhs", ["depth", "maxval(depth(:))"])
-@pytest.mark.xfail(reason="#717 need to distinguish scalar and array "
-                   "assignments")
-def test_where_with_scalar_assignment(rhs):
+def test_where_with_scalar_assignment():
     ''' Test that a WHERE containing a scalar assignment is handled correctly.
     Currently it is not as we do not distinguish between a scalar and an array
     reference that is missing its colons. This will be fixed in #717.
```

Comment:
I am confused by this, do we handle this correctly now or not? If not, shouldn't it be a CodeBlock?

I think I would refactor this test to be: "test_where_with_plain_references" where we have 3 test cases with the same code and a plain reference in it.
In one case the reference is an scalar.
In the second the reference is an array without array syntax.
In the thirst is undecidable (e.g. comes from an import)

The first should keep it as scalar, the second should transform it to an indexed array and the third should put it in a codeblock. 

Also, its's up to you but I prefer to skip the process_where and use a full fortran_reader.psyir_from_source('code') snippet, where it is easier to see how each variable is declared, even if it is a bit longer.

Reply:
Yeah this comment was out of date.
I'll update the test to use fortran_reader too.

Reply:
I also don't really understand if this test is valid fortran - can you check if it makes sense?

Reply:
I think it does provided all the arrays are the same size in all dimensions (as i defined)